scan-command/src/main/java/io/ballerina/scan/internal/CoreRule.java

```diff
@@ -31,8 +31,9 @@
  * */
 enum CoreRule {
     AVOID_CHECKPANIC(RuleFactory.createRule(1, "Avoid checkpanic", RuleKind.CODE_SMELL)),
-    UNUSED_FUNCTION_PARAMETER(RuleFactory.createRule(2,
-            "Unused function parameter", RuleKind.CODE_SMELL));
+    UNUSED_FUNCTION_PARAMETER(RuleFactory.createRule(2, "Unused function parameter", RuleKind.CODE_SMELL)),
+    PUBLIC_NON_ISOLATED_CONSTRUCT(RuleFactory.createRule(3,
+            "Non isolated public class or function/method", RuleKind.CODE_SMELL));
```

Comment:
Can we break this down to two rules? Non-isolated object and non-isolated function/method.